src/main/scala/com/cognite/spark/datasource/EventsRdd.scala

```diff
@@ -0,0 +1,59 @@
+package com.cognite.spark.datasource
+
+import com.softwaremill.sttp.Uri
+import org.apache.spark.{InterruptibleIterator, Partition, SparkContext, TaskContext}
+import org.apache.spark.rdd.RDD
+import org.apache.spark.sql.Row
+import io.circe.generic.auto._
+
+case class EventsRddPartition(
+    cursor: Option[String],
+    size: Option[Int],
+    index: Int,
+    filter: EventFilter)
+    extends Partition
+
+case class EventsRdd(
+    @transient override val sparkContext: SparkContext,
+    getSinglePartitionBaseUri: Uri,
+    toRow: EventItem => Row,
+    eventFilters: Array[EventFilter],
+    config: RelationConfig,
+    cursors: Iterator[(Option[String], Option[Int])])
+    extends RDD[Row](sparkContext, Nil)
+    with CdpConnector {
+
+  override def getPartitions: Array[Partition] =
+    (for {
+      cursor <- cursors
+      eventFilter <- eventFilters
+    } yield (cursor, eventFilter)).zipWithIndex.map {
+      case (((cursor, size), eventFilter), index) =>
+        val partitionSize = (config.limit, size) match {
```

Comment:
This is taken care of in `Batch.withCursor`, or maybe somewhere else, I think...? In this particular case, however, "size" will always be set to None since the cursors are from the cursors endpoint.